consoleme/lib/dynamo.py

```diff
@@ -524,11 +535,50 @@ async def get_policy_requests(self, arn=None, request_id=None):
         matching_requests = []
         if requests["Items"]:
             items = self._data_from_dynamo_replace(requests["Items"])
+            items = await self.convert_policy_requests_to_v3(items)
             matching_requests.extend(items)
         return matching_requests
 
+    async def convert_policy_requests_to_v3(self, requests):
+        # Remove this function and calls to this function after a grace period of
+        changed = False
+        for request in requests:
+            if not request.get("version") in ["2"]:
+                continue
+            if request.get("extended_request") and not request.get("principal"):
+                principal_arn = request.pop("arn")
+                request["principal"] = {
+                    "principal_arn": principal_arn,
+                    "principal_type": "AwsResource",
+                }
+                request["extended_request"]["principal"] = {
+                    "principal_arn": principal_arn,
+                    "principal_type": "AwsResource",
+                }
+            request.pop("arn", None)
+            changes = (
+                request.get("extended_request", {})
+                .get("changes", {})
+                .get("changes", [])
+            )
+            for change in changes:
+                if not change.get("principal_arn"):
+                    continue
+                if not change.get("version") in ["2.0", 2]:
```

Comment:
Does this also need a string-2, i.e. `["2.0", "2", 2]`?

Reply:
Why, yes we do. I just literally checked all of the unique change versions we have and '2' is one of them. Nice catch!